pkg/config/config_parser.go

```diff
@@ -169,28 +167,37 @@ func decodeConfigYAML(r io.Reader) (*Config, diag.Diagnostics) {
 		return nil, diags
 	}
 
+	providers := yc.Providers
+	diags := diag.Diagnostics{}
+	for _, p := range providers {
+		p.Configuration, err = yaml.Marshal(p.ConfigKeys["configuration"])
+		if err != nil {
+			diags = diags.Add(diag.FromError(err, diag.INTERNAL, diag.WithSummary("ConfigKeys marshal failed")))
+			continue
+		}
+	}
+
 	c := &Config{
 		CloudQuery: yc.CloudQuery,
+		Providers:  providers,
 		format:     cqproto.ConfigYAML,
 	}
 
+	return c, nil
+}
+
+func (p *Parser) decodeConfigHCL(name string, data []byte) (*Config, diag.Diagnostics) {
```

Comment:
This function is designated to be removed https://github.com/cloudquery/cloudquery/pull/1044